src/services/fields.ts

```diff
@@ -171,3 +187,29 @@ export function getLabelTypeFromFrame(labelKey: string, frame: DataFrame, index
       return null;
   }
 }
+
+export function getFilterTypeFromLabelType(type: LabelType | null, key: string, value: string): VariableFilterType {
+  switch (type) {
+    case LabelType.Indexed: {
+      return VAR_LABELS;
+    }
+    case LabelType.Parsed: {
+      return VAR_FIELDS;
+    }
+    case LabelType.StructuredMetadata: {
```

Comment:
I don't understand this. LEVELS for structured metadata?

Reply:
Ok, it wasn't as bad as I thought last night, I think this was carrying over the old assumption before this branch that the only structured metadata was `detected_level`. Updated to only add `detected_level` as a level, any other structured metadata must be a field variable.

Might be nice to update the level variable in the future to just hold generic structured metadata (or create a new variable for it), but for now this should suffice.

Reply:
Right. The whole point of VAR_LEVELS is to be able to isolate them from the Logs Volume query and maybe other tiny extra magic with the Volume filters, but we might (in the future) achieve the same by looking for that key somewhere else.